CalibTracker/SiStripESProducers/plugins/real/SiStripBackPlaneCorrectionDepESProducer.cc

```diff
@@ -16,57 +16,85 @@
+#include "CondFormats/SiStripObjects/interface/SiStripBackPlaneCorrection.h"
+#include "CondFormats/SiStripObjects/interface/SiStripLatency.h"
+#include "CalibTracker/Records/interface/SiStripDependentRecords.h"
+
+class SiStripBackPlaneCorrectionDepESProducer : public edm::ESProducer {
+ public:
+  SiStripBackPlaneCorrectionDepESProducer(const edm::ParameterSet&);
+  ~SiStripBackPlaneCorrectionDepESProducer() override{};
   
-  edm::LogInfo("SiStripBackPlaneCorrectionDepESProducer") << "ctor" << std::endl;
+  std::unique_ptr<SiStripBackPlaneCorrection> produce(const SiStripBackPlaneCorrectionDepRcd&);
+   
+ private:
+  edm::ESGetToken<SiStripLatency, SiStripLatencyRcd> latencyToken_;
+  edm::ESGetToken<SiStripBackPlaneCorrection, SiStripBackPlaneCorrectionRcd> backPlaneCorrectionPeakToken_;
+  edm::ESGetToken<SiStripBackPlaneCorrection, SiStripBackPlaneCorrectionRcd> backPlaneCorrectionDeconvToken_;
+};
+
+SiStripBackPlaneCorrectionDepESProducer::SiStripBackPlaneCorrectionDepESProducer(const edm::ParameterSet& iConfig) {
+  auto cc = setWhatProduced(this);
+
+  edm::LogInfo("SiStripBackPlaneCorrectionDepESProducer") << "ctor";
+
+  auto getLatency = iConfig.getParameter<edm::ParameterSet>("LatencyRecord");
+  // How useful the "record" parameter really is?
+  if(getLatency.getParameter<std::string>("record") == "SiStripLatencyRcd") {
+    // Shouldn't the label be a tracked parameter?
+    cc.setConsumes(latencyToken_, edm::ESInputTag{"", getLatency.getUntrackedParameter<std::string>("label")});
+  }
+  // Would it make sense to elevate this as an exception?
+  else edm::LogError("SiStripBackPlaneCorrectionDepESProducer") << "[SiStripBackPlaneCorrectionDepESProducer::ctor] No Latency Record found ";
 
+  auto getPeak = iConfig.getParameter<edm::ParameterSet>("BackPlaneCorrectionPeakMode");
+  if(getPeak.getParameter<std::string>("record") == "SiStripBackPlaneCorrectionRcd") {
+    // Shouldn't the label be a tracked parameter?
+    cc.setConsumes(backPlaneCorrectionPeakToken_, edm::ESInputTag{"", getPeak.getUntrackedParameter<std::string>("label")});
+  // Would it make sense to elevate this as an exception?
+  } else edm::LogError("SiStripBackPlaneCorrectionDepESProducer") << "[SiStripBackPlaneCorrectionDepESProducer::ctor] No Lorentz Angle Record found " << std::endl;
+
+
+  auto getDeconv = iConfig.getParameter<edm::ParameterSet>("BackPlaneCorrectionDeconvMode");
+  // How useful the "record" parameter really is?
+  if(getDeconv.getParameter<std::string>("record") == "SiStripBackPlaneCorrectionRcd") {
+    // Shouldn't the label be a tracked parameter?
+    cc.setConsumes(backPlaneCorrectionDeconvToken_, edm::ESInputTag{"", getDeconv.getUntrackedParameter<std::string>("label")});
+  // Would it make sense to elevate this as an exception?
+  } else edm::LogError("SiStripBackPlaneCorrectionDepESProducer") << "[SiStripBackPlaneCorrectionDepESProducer::ctor] No Lorentz Angle Record found " << std::endl;
```

Comment:
I have a few general questions (same apply for SiStripLorentzAngleDepESProducer below but I won't repeat)
* Shouldn't the parameters be tracked?
* Is the `record` parameter really useful?
* In case the check on the `record` is kept, would it make sense to throw an exception instead (for incorrect configuration)?

Bu quick look this class and SiStripLorentzAngleDepESProducer share all the code except types being different, so in principle one could use class template here (though it's probably not worth to do that now).

@mmusich @tsusa

Reply:
I won't be able to look at this until next week. Can we put this on hold?

Reply:
That's fine. (the PR is already sort-of-holded by being in "draft" status)